### PR TITLE
Moved file existence checks to FileLoader, added an option to disable it on class level

### PR DESCRIPTION
**Pull request by szymach**

### CODE REVIEW

File: src/Exception/FileNotFoundException.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+declare(strict_types=1);
+
+namespace FSi\Component\Files\Exception;
+
+use Exception;
+use FSi\Component\Files\WebFile;
+
+use function sprintf;
+
+final class FileNotFoundException extends Exception
```
Comment (chives) [inline src/Exception/FileNotFoundException.php:19]: `Exception` -> `RuntimeException` or even better create some package-specific root exception class (inheriting from `RuntimeException`) or interface

File: src/Entity/FileLoader.php [hunk 2]
```diff
@@ -62,6 +75,73 @@ public function fromEntity(FilePropertyConfiguration $configuration, object $ent
...
-        return $this->fileManager->load($configuration->getFileSystemName(), $path);
+        $file = $this->fileManager->load($configuration->getFileSystemName(), $path);
+        if (null !== $file) {
+            $this->checkFileExistenceIfEnabled($className, $file);
+        }
+
+        return $file;
+    }
+
+    public function disableFileExistanceChecksOnLoad(): void
```
Comment (chives) [inline src/Entity/FileLoader.php:86]: this method with misspelled name can be removed

File: src/Entity/FileLoader.php [hunk 3]
```diff
@@ -62,6 +75,73 @@ public function fromEntity(FilePropertyConfiguration $configuration, object $ent
...
+    {
+        $this->disableFileExistenceChecksOnLoad();
+    }
+
+    public function enableFileExistanceChecksOnLoad(): void
```
Comment (chives) [inline src/Entity/FileLoader.php:91]: this method with misspelled name can be removed

File: src/FileManagerConfigurator/FileExistenceChecksConfigurator.php [hunk 4]
```diff
@@ -15,4 +15,6 @@ interface FileExistenceChecksConfigurator extends FileExistanceChecksConfigurato
 {
     public function disableFileExistenceChecksOnLoad(): void;
     public function enableFileExistenceChecksOnLoad(): void;
+    public function disableClassFileExistenceChecksOnLoad(string $className): void;
+    public function enableClassFileExistenceChecksOnLoad(string $className): void;
```
Comment (chives) [inline src/FileManagerConfigurator/FileExistenceChecksConfigurator.php:19]: consider removing `OnLoad` from method names

File: src/Integration/Symfony/DependencyInjection/FilesExtension.php [hunk 5]
```diff
@@ -34,6 +34,7 @@ final class FilesExtension extends Extension
      *   entities: array<string, array{
      *     filesystem: string,
      *     prefix: string,
+     *     disableFileChecksOnLoad: bool,
```
Comment (chives) [inline src/Integration/Symfony/DependencyInjection/FilesExtension.php:37]: does this option even exist?

Comment (szymach) [reply src/Integration/Symfony/DependencyInjection/FilesExtension.php:37]: I think you checked an older commit

File: src/Entity/FileLoader.php (hunk 3 above)
Comment (szymach) [reply src/Entity/FileLoader.php:91]: These checks are turned off only during loading files, so without it it would be confusing.

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

**Review by chives — APPROVED** [review]

**Review by rn0 — APPROVED** [review]

**Pull request merged**